src/ir/builtins.py

```diff
@@ -138,9 +138,19 @@ def get_function_types(self, max_parameters):
     def get_nothing(self):
         raise NotImplementedError
 
-    def get_constant_candidates(self):
+    def get_dynamic_types(self, gen_object):
```

Comment:
I don't like the name `dynamic_types`. Can we think of a better name? `compound types` maybe?

Reply:
Done!